include/deal.II/base/segment_data_out.h

```diff
@@ -0,0 +1,160 @@
+// ------------------------------------------------------------------------
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+// Copyright (C) 2025 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// Part of the source code is dual licensed under Apache-2.0 WITH
+// LLVM-exception OR LGPL-2.1-or-later. Detailed license information
+// governing the source code and code contributions can be found in
+// LICENSE.md and CONTRIBUTING.md at the top level directory of deal.II.
+//
+// ------------------------------------------------------------------------
+
+#ifndef dealii_segment_data_out_h
+#define dealii_segment_data_out_h
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/data_out_base.h>
+#include <deal.II/base/point.h>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * This class generates graphical output for line segment objects.
+ *
+ * Line segments are one-dimensional objects embedded in a @p dim dimensional
+ * space. This class creates output data that can be visualized as lines in
+ * visualization tools. Each segment is represented as a pair of Point<dim>
+ * objects defining the start and end points.
+ */
+template <int dim>
+class SegmentDataOut : public DataOutInterface<1, dim>
+{
+public:
+  /**
+   * Constructor.
+   */
+  SegmentDataOut() = default;
+
+  /**
+   * Destructor.
+   */
+  ~SegmentDataOut() = default;
+
+  /**
+   * Generate patches from a vector of line segments.
+   *
+   * @param segments A vector of line segments, where each segment is represented
+   *        as a std::pair of two Point<dim> objects (start and end points).
+   */
+  void
+  build_patches(const std::vector<std::pair<Point<dim>, Point<dim>>> &segments);
+
+  /**
+   * Attach data to each output patch that was generated by build_patches().
+   *
+   * The @p datasets parameter is expected to have the same size of the
+   * container you used in the call to build_patches(), and each entry should
+   * have the same size of the @p dataset_names argument.
+   *
+   * @param[in] datasets The actual data to attach to each patch
+   * @param[in] dataset_names The name of each component of the dataset
+   */
+  void
+  add_datasets(const std::vector<std::vector<double>> &datasets,
+               const std::vector<std::string>         &dataset_names);
```

Comment:
I'm wondering why you split this functionality into two functions. Is it that one can call `add_datasets` more than once? If you can call it only once, it would seem simpler to just combine the two functions into `build_patches()` that takes two additional arguments one can leave empty if one wants.